solr/solrj/src/java/org/apache/solr/client/solrj/impl/CloudHttp2SolrClient.java

```diff
@@ -32,10 +32,6 @@
  * Solr endpoints for SolrCloud collections, and then use the
  * {@link LBHttp2SolrClient} to issue requests.
  *
- * This class assumes the id field for your documents is called
- * 'id' - if this is not the case, you must set the right name
```

Comment:
the latter part can go but isn't the first part of this still sound -- that we assume "id"?

Reply:
Hmm, tricky. So the https://github.com/apache/lucene-solr/commit/f9410e14614d1f9ebb88226c287f38cd4d0b3d1f commit removes the `idField = ID` and replaces it with `routeFieldDeprecated = null` i.e. no more use of "id" there but ...

When it comes to `routeFieldDeprecated` usage at https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.7.0/solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java#L526-L527 we have
```
...
String routeField = (routeFieldDeprecated != null) ? routeFieldDeprecated :
        (col.getRouter().getRouteField(col) == null) ? ID : col.getRouter().getRouteField(col);
...
```
i.e. yes in conjunction with the "the field is automatically determined from Zookeeper" comment at https://github.com/apache/lucene-solr/blob/releases/lucene-solr%2F8.7.0/solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java#L311 there is an "id" assumption of sorts, though in most use cases the field would be automatically determined from ZooKeeper I guess?

Reply:
What is coming from ZooKeeper?  Maybe the "route field" which is in the collection state (DocCollection) but not the uniqueKey field which is in the schema and AFAIK it not resolved by CloudSolrClient.